Sources/Player/OfflineEngine/Internal/Storage/PlayableOfflinedProduct.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+struct PlayableOfflinedProduct: Equatable {
+	let productType: ProductType
+	let productId: String
+	let assetPresentation: AssetPresentation
+	let audioMode: AudioMode?
+	let audioQuality: AudioQuality?
+	let audioCodec: AudioCodec?
+	let audioSampleRate: Int?
+	let audioBitDepth: Int?
+	let videoQuality: VideoQuality?
+	let mediaUrl: URL
+	let licenseUrl: URL?
+	let albumReplayGain: Float?
+	let albumPeakAmplitude: Float?
+	let trackReplayGain: Float?
+	let trackPeakAmplitude: Float?
+
+	init?(from offlineEntry: OfflineEntry?) {
```

Comment:
`OfflineEntry` can be non-optional, good call.

But we can't make `mediaURL` non-optional because resolving the bookmark data to a URL can be nil. (So, we store the bookmarkData representing the URL in the database, and when we retrieve it at a later date and try to form a URL, it can be nil / no longer valid)